tpke/src/lib.rs

```diff
@@ -492,16 +476,158 @@ mod tests {
+        let rng = &mut test_rng();
+        let shares_num = 16;
+        let threshold = shares_num * 2 / 3;
+
+        let (_, _, mut contexts) =
+            setup_simple::<E>(threshold, shares_num, rng);
+
+        // Prepare participants
+
+        // First, save the soon-to-be-removed participant
+        let selected_participant = contexts.pop().unwrap();
+        let x_r = selected_participant
+            .public_decryption_contexts
+            .last()
+            .unwrap()
+            .domain;
+        let original_y_r =
+            selected_participant.private_key_share.private_key_shares[0];
+
+        // Now, we have to remove the participant from the contexts and all nested structures
+        let mut remaining_participants = contexts;
+        for p in &mut remaining_participants {
+            p.public_decryption_contexts.pop();
+        }
+
+        // Refresh the share
+        let y_r = recover_share_at_point(
+            &remaining_participants,
+            threshold,
+            &x_r,
+            rng,
+        );
+        assert_eq!(y_r.into_affine(), original_y_r);
+    }
+
+    #[test]
+    fn simple_threshold_decryption_with_share_recovery_at_point() {
+        let rng = &mut test_rng();
+        let shares_num = 16;
+        let threshold = shares_num * 2 / 3;
+        let msg: &[u8] = "abc".as_bytes();
+        let aad: &[u8] = "my-aad".as_bytes();
+
+        let (pubkey, _, contexts) =
+            setup_simple::<E>(threshold, shares_num, rng);
+        let ciphertext = encrypt::<_, E>(msg, aad, &pubkey, rng);
+
+        // Remove one participant from the contexts and all nested structures
+        let mut remaining_participants = contexts;
+        remaining_participants.pop();
+        for p in &mut remaining_participants {
+            p.public_decryption_contexts.pop();
+        }
+
+        // Refresh the share
+        let x_r = Fr::rand(rng);
+        let y_r = recover_share_at_point(
+            &remaining_participants,
+            threshold,
+            &x_r,
+            rng,
+        );
+        let recovered_key_share = PrivateKeyShare {
+            private_key_shares: vec![y_r.into_affine()],
+        };
+
+        // Creating decryption shares
+        let mut decryption_shares: Vec<_> = remaining_participants
+            .iter()
+            .map(|ctxt| {
+                make_decryption_share(&ctxt.private_key_share, &ciphertext)
+            })
+            .collect();
+        decryption_shares
+            .push(make_decryption_share(&recovered_key_share, &ciphertext));
+
+        // Creating a shared secret from remaining shares and the recovered one
+        let shares_x = &remaining_participants[0]
+            .public_decryption_contexts
+            .iter()
+            .map(|ctxt| ctxt.domain)
+            .collect::<Vec<_>>();
+        let lagrange = prepare_combine_simple::<E>(shares_x);
+
+        let shared_secret =
```

Comment:
This name is confusing. The ferveo paper call this `ephemeral shared secret`, but I think the word "shared" is misleading and doesn't add anything. Let's rename this to `ephemeral secret`.

This is important to avoid confusion with the real shared secret (i.e. the distributed secret after the DKG protocol)